### DISCUSSION TITLE
Take off simulation on JSBSim

### DISCUSSION THREAD

**Discussion by GS-Campos** [JSBSim-Team/jsbsim#663] [General]
Hi guys, 

I'm trying to simulate a take off on JSBSim and I have some questions about it. 
1- When I do some command to engine, independently of this command is a step or a ramp, appears a sinusoidal behavior in gamma. Exist a right way to start the engine on this case?
2- I created a lower radome as structure on ground reactions to predicted if this structure touch the ground during the take off. When I look the outputs, exist a compressive force actuating on this structure, even it not touching the floor. These forces actuating on structure are calculated as the same way of the bogey types (F = kx + bu)? How can I check if the structure touch the ground or not?

Thanks

**Comment by seanmcleod** [comment]
For 1 are you not exciting the Phugoid mode?

**Reply by agodemar** [reply]
@GS-Campos what is your background? Are you aware of the aircraft response modes?

**Reply by GS-Campos** [reply]
Hi Agostino and Sean. Thank you for yours contributions. I understand the modes, but I don't think on them because it's occurred on the ground, immediately when I start the engine.
I will provide more information now:
* First, I set some commands like flap, the position of horizontal tail (to trimmed the airplane at the V2 after take off). Thus, I put the maximum throttle on engines and, when the VR is hit, the elevator is triggered:
![image](https://user-images.githubusercontent.com/107955673/175016168-f4e9950e-44e5-43fb-8bc5-d61f5383be1a.png)

The pike who I has comment is show below:
![image](https://user-images.githubusercontent.com/107955673/175017468-7118eb5f-8956-478a-87db-058401cb8b63.png)
I tried to use the brakes and this peak stayed.

I would to know: this peak is normal? Is it possible to avoid this? Is it a numerical error?

Thanks

**Comment by seanmcleod** [comment]
@GS-Campos rather use triple backticks ` followed by xml and then paste the xml from the script file followed by triple backticks to format xml, python, etc. snippets rather than pasting an image.

In terms of the early spike in the first second or so how are you placing the aircraft on the ground? If you set the altitude of the aircraft to 0 for example then you are setting the cg to be at 0, which means that the gear are then extremely compressed and then bounce back and you get these sorts of oscillations.

**Reply by GS-Campos** [reply]
I'm sorry @seanmcleod 
```xml    
<event name="Set the ic">
	    <condition> simulation/sim-time-sec le 0.01 </condition> 
		<set name="aero/alpha-max-rad" value="-0.262" />
		<set name="aero/alpha-min-rad" value="-0.262" />
		<set name="aero/blue" value = "1" />
		<set name="fcs/flap-pos-deg" value="6"/>	
		<set name="fcs/pitch-trim-cmd-norm" value="-0.45"/>			
    <notify/>
    </event>		
	
    <event name="Set engines running">
	    <condition> simulation/sim-time-sec le 0.02 </condition> 		
        <set name="propulsion/engine[0]/set-running" value="1"/>
        <set name="propulsion/engine[1]/set-running" value="1"/>			
        <set name="fcs/throttle-cmd-norm[0]" value="1"/>
        <set name="fcs/throttle-cmd-norm[1]" value="1"/>	
		<!-- <set name="fcs/pitch-trim-cmd-norm" value="-0.661"/>		 -->
		<!-- <set name="fcs/right-brake-cmd-norm" value="0"/> -->
        <!-- <set name="fcs/left-brake-cmd-norm" value="0"/> -->
    <notify/>
    </event>	

  <event name="Take-off" >
		<condition>	velocities/vc-kts ge 128.5 </condition>
		<set name="fcs/TrimWithHTorEle" value="1"/>
		<set name="fcs/pitch-trim-cmd-norm" value="-0.125"/>	
		<set name="fcs/elevator-cmd-norm" value="-0.25" action="ramp" tc="2.5"/>		
	<notify/>
   </event>
```
Interesting about the ground. I'm starting the airplane on ground with all velocities and angles zero:
```\xml
  <altitude unit="FT">   0</altitude>
  <vt unit="KTS">       0 </vt>
  <gamma unit="DEG">       0.0 </gamma>
  <latitude unit="DEG">    0.0 </latitude>
  <longitude unit="DEG">   0.0 </longitude>
  <phi unit="DEG">         0 </phi>
  <psi unit="DEG">         0.0 </psi>
  <beta unit="DEG">        0 </beta>
```

**Reply by agodemar** [reply]
@GS-Campos consider also that  $\alpha $ and  $\gamma $ have have not much sense at a low, near-zero speed. This happens in the first few seconds of your simulation. Those wild oscillations are also due, probably, to the gear compression.

**Comment by seanmcleod** [comment]
> I'm starting the airplane on ground

But that was my point, which part of the plane do you think is on the ground at 0ft?

With the cg at 0ft what happens to the gear? They're super compressed and given the spring parameters for the gear it means they generate a massive force.

**Reply by agodemar** [reply]
@GS-Campos that's the reason why the $\alpha $ grows negatively and  $\gamma $ positively. The airframe is pushed upwards

**Comment by seanmcleod** [comment]
Take a look at the relevant snippets from the 737 and it's reset file for starting on the ground.

https://github.com/JSBSim-Team/jsbsim/blob/0e842e751373f70ac927cef87a68a6e8fd1129df/aircraft/737/737.xml#L64-L68

https://github.com/JSBSim-Team/jsbsim/blob/0e842e751373f70ac927cef87a68a6e8fd1129df/aircraft/737/737.xml#L72-L77

https://github.com/JSBSim-Team/jsbsim/blob/0e842e751373f70ac927cef87a68a6e8fd1129df/aircraft/737/reset00.xml#L2-L17

**Comment by seanmcleod** [comment]
So in the 737 example you can see that the gear are 44 inches below the **empty weight** cg. Once loaded with fuel etc. the cg will be slightly different, you can see what the final cg is by looking at the mass properties report, e.g.

```sh
 C:\source\jsbsim>.\Release\JSBSim.exe --script=scripts\B737_Runway.xml
```

```sh
  Mass Properties Report (English units: lbf, in, slug-ft^2)
                                      Weight    CG-X    CG-Y    CG-Z         Ixx         Iyy         Izz         Ixy         Ixz         Iyz
    Base Vehicle                       83000.0   639.0     0.0   -40.0    562000.0   1473000.0   1894000.0        -0.0      8000.0        -0.0
0   Fuel                                 10000     520     -80     -18           0           0           0
1   Fuel                                 10000     520      80     -18           0           0           0
2   Fuel                                  4000     480       0     -18           0           0           0


    Total:                          107000.0   610.8     0.0   -35.1    591572.3   1539552.7   1986235.4         0.0     19109.1         0.0

End of vehicle configuration loading.
```

So you can see that the final `CG-Z` is -35.1 inches.

Now in general you can use that information to fine-tune your starting/initial altitude so that there is close to no gear compression and/or very little drop from a height onto the gear causing a very slight bounce, which will really diminish the sorts of spikes you're seeing, but you will still need to ignore the very minor spikes that remain. Or you can initiate a `ground trim` and allow JSBSim to get the aircraft perfectly settled on the ground.

**Reply by GS-Campos** [reply]
You are right. The problem was the altitude. I was considering the CG on Z = 0 and the landing gear in -117. But, i initiate the airplane in altitude 0. When I put the airplane in a higher altitude the peaks are less than before. Thanks @seanmcleod  and @agodemar

**Comment by GS-Campos** [comment]
And the other thing is: I have a airplane with lower radome and I want to prevent if it touch the ground during the take off. I declared it like a structure:
```xml
         <contact type="STRUCTURE" name="Lower Radome">
            <location unit="IN">
                <x> 627.56 </x> 
                <y> 0 </y>
                <z> -71.73 </z> 		

            </location>
            <static_friction> .75 </static_friction> <!-- dummy -->
            <dynamic_friction> .5 </dynamic_friction> <!-- dummy -->
            <spring_coeff unit="LBS/FT">       1830.15 </spring_coeff> <!-- dummy -->
            <damping_coeff unit="LBS/FT/SEC">   366.03 </damping_coeff> <!-- dummy -->
        </contact>
```
The displecement is in relation of the structural frame, right? Thus, I need to verify using this displecement or exist a warning mensage? Exisit another way to declareted rigid strucutures? Because I'm a little confused due to the spring and damping coefficients.

Thanks

**Reply by bcoconni** [reply]
Your best course of action is to check if the flag `contact/unit[*]/WOW` is nonzero (you need to replace the star `*` with the proper index of your `<contact>` element). Whenever the flag is different from zero then this means that the point is on contact with ground.
